src/ReadonlyArray.ts

```diff
@@ -1649,7 +1649,37 @@ export const partitionMap: {
  * @category filtering
  * @since 2.0.0
  */
-export const compact: <A>(self: Iterable<Option<A>>) => Array<A> = filterMap(identity)
+export const filterSome: <A>(self: Iterable<Option<A>>) => Array<A> = filterMap(identity)
+
+/**
+ * @category filtering
+ * @since 2.0.0
+ */
+export const filterLeft = <E>(self: Iterable<Either<E, any>>): Array<E> => {
```

Comment:
```suggestion
export const filterLeft = <E, A>(self: Iterable<Either<E, A>>): Array<E> => {
```

Reply:
@tim-smart just for my understanding: whats the benefit of capturing `A` if we are eliminating it?

Reply:
We have run into subtle issues with inference and assignability before.

So where possible we just leave unused generics unused. They might be called `X` in some places to indicate that they are not used.

Reply:
Ok. I'd be curious to know more about that. Happy to make the adjustment now but I'm always curious to understand these edge cases with typescript ;-)

Reply:
It's due to the TS inference algorithm that prefers weaker types as a shortcut, basically having `any` here will make TS back-infer to `any` the value of `A` in for example `pipe(a, filterLeft(f))`, we had hard to spot issues because of this kind of stuff so now we always put a generic even if ignored

Reply:
Ah ok. That actually makes sense. It's a bit quirky but kinda makes sense ...